src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp

```diff
@@ -7557,6 +7559,125 @@ void mlir::transform::addConcatenateOpCanon(RewritePatternSet &patterns,
   patterns.insert<ConcatenateOpCanon>(maxConstantExpansion, &context, benefit);
 }
 
+///////////////  End stablehlo patterns
+
+struct RemoveIVs : public OpRewritePattern<scf::ForOp> {
```

Comment:
we should put this in a separate pass and leave this one for just HLO ops

Reply:
well the reason why I put it here it's because we have all the infra for writing unit tests for pattern that can be reused.
So the suggestion is to write a pass instead of a pattern here?

Reply:
Alright I've put out a test transform framework for testing these kinds of patterns